### PR TITLE
hourofcode.com homepage hero redesign

### PR DESCRIPTION
**Pull request by tanyaparker**
**Before (not playing video)**
screen shot 2018-10-22 at 4 49 17 pm

**Before (playing video)**
screen shot 2018-10-22 at 4 16 43 pm

**After (desktop**
screen shot 2018-10-22 at 5 18 15 pm

**After (mobile)**
screencapture-localhost-hourofcode-3000-us-2018-10-22-17_18_42

**hourofcode.com/map normal view** 
screen shot 2018-10-22 at 5 07 44 pm

**hourofcode.com/map viewing from a country page**
screen shot 2018-10-22 at 5 08 38 pm

**hourofcode.com/map viewing from a company page** 
screen shot 2018-10-22 at 5 08 11 pm

### CODE REVIEW

File: pegasus/sites.v3/hourofcode.com/views/announcements.haml [hunk 1]
```diff
@@ -5,10 +5,20 @@
   providers << {id: "kano", url: "https://kano.me/"}
   providers << {id: "thimble", url: "https://www.thimble.io/"}
   providers << {id: "dexter", url: "https://www.gigglebot.io/"}
+  latam_language_codes = ["la", "es", "pt", "po"]
+  show_latam_videos = DCDO.get("show_latam_videos", nil)
```
Comment (tanyaparker) [inline pegasus/sites.v3/hourofcode.com/views/announcements.haml]: I was thinking about removing this but I think it's still up in the air if we'll want to show the English creativity video to all audiences during MC and Dance Party launches. We can remove after HOC 2018 if we've decided it's permanent.

File: pegasus/sites.v3/hourofcode.com/views/hoc_events_counter.haml [hunk 2]
```diff
@@ -6,31 +6,15 @@
 -if @company
   -company_name = hoc_s("company_" + @company)
   -company_total = format_integer_with_commas(company_count)
-
-  %h2.counter-title
-    %a{href: resolve_url('/events/all')}
-      = events_text + ","
-      = hoc_s(:map_header_company).gsub('#', company_total).gsub('company', company_name)
-    -# Hide signup buttons during Hour of Code
```
Comment (tanyaparker) [inline pegasus/sites.v3/hourofcode.com/views/hoc_events_counter.haml:14]: We used to show a "Sign up" button to the right of the map title. I was thinking of putting it under `hoc_mode` but it won't make sense to have a "Sign up" button after this text with this new design.

File: pegasus/sites.v3/hourofcode.com/i18n/en.yml [hunk 3]
```diff
@@ -313,7 +316,7 @@
 
   map_title: "Hour of Code map"
   map_header: '# events registered for Hour of Code 2017'
-  map_header_count_year: '%{event_count} events registered for Hour of Code %{year}'
+  map_header_count_year: '%{event_count} events registered in 2018'
```
Comment (breville) [inline pegasus/sites.v3/hourofcode.com/i18n/en.yml]: Can we still use `%{year}` here?

File: pegasus/sites.v3/hourofcode.com/i18n/en.yml [hunk 4]
```diff
@@ -19,13 +19,16 @@
   codeorg_homepage_hoc2018_header_line2_mobile: 'What will you create?'
   codeorg_homepage_hoc2018_header_line3: 'Give flight to imagination and birth to innovation. Start with an Hour of Code and your own brand of creativity, whatever it may be.'
   codeorg_homepage_hoc2018_video: 'Watch our video'
+  hourofcode_homepage_hoc2018_header_line1: 'Anyone, anywhere can organize an Hour of Code event. One-hour tutorials in over 45 languages. No experience needed.'
+  hourofcode_homepage_hoc2018_header_line2: 'Ages 4 to 104.'
```
Comment (Erin007) [inline pegasus/sites.v3/hourofcode.com/i18n/en.yml:24]: teeny tiny nit: Does it make sense to put these strings in order: hourofcode_homepage_hoc2018_header_line1, then hourofcode_homepage_hoc2018_header_line2, then codeorg_homepage_hoc2018_header_line3 (currently on line 20)?

File: pegasus/sites.v3/hourofcode.com/public/map.haml [hunk 5]
```diff
@@ -7,6 +7,6 @@ layout: wide_index
 %script{type: "text/javascript", src: "/js/jquery.geocomplete.min.js"}
 
 = view :header
-= view :hoc_events_counter
+%h2{style: "text-align: center;"}= view :hoc_events_counter
```
Comment (Erin007) [inline pegasus/sites.v3/hourofcode.com/public/map.haml]: Is there a css file for this already? Can we pull these in-line styles and add there if so?

File: pegasus/sites.v3/hourofcode.com/views/announcements.haml (hunk 1 above)
Comment (Erin007) [reply pegasus/sites.v3/hourofcode.com/views/announcements.haml]: Sounds good to me.

File: pegasus/sites.v3/hourofcode.com/i18n/en.yml (hunk 4 above)
Comment (breville) [reply pegasus/sites.v3/hourofcode.com/i18n/en.yml:24]: Aren't they separate sets?

File: pegasus/sites.v3/hourofcode.com/public/css/front-page.css [hunk 6]
```diff
@@ -81,11 +105,39 @@ a.category-link {
...
   text-shadow: 0px 0px 4px rgba(31, 31, 31, 1.0);
+  text-align: center;
+}
+
+.front-header-banner {
+  color: white;
+  margin-top: 0px;
+}
+
+.front-header-banner h4{
```
Comment (breville) [inline pegasus/sites.v3/hourofcode.com/public/css/front-page.css]: tiny nit: space before `{`

**Comment by breville** [comment]
Does it look good all the way down at 320px width too?

File: pegasus/sites.v3/hourofcode.com/public/index.haml [hunk 7]
```diff
@@ -224,52 +201,7 @@ social:
     // hoc2018: temporarily disable the cycling.
     //timerId = setInterval(cycleImage, 6000);
   });
-
-  function startVideo()
```
Comment (breville) [inline pegasus/sites.v3/hourofcode.com/public/index.haml:228]: It's nice to get rid of so much code, but do we think we'll be restoring it sooner or later?

**Comment by breville** [comment]
What exactly changed in `/map`?

File: pegasus/sites.v3/hourofcode.com/i18n/en.yml (hunk 3 above)
Comment (tanyaparker) [reply pegasus/sites.v3/hourofcode.com/i18n/en.yml]: Oops. Yes!

File: pegasus/sites.v3/hourofcode.com/i18n/en.yml (hunk 4 above)
Comment (tanyaparker) [reply pegasus/sites.v3/hourofcode.com/i18n/en.yml:24]: Brendan is correct about my intention. The first set is for the code.org homepage and the second send is for the hourofcode.com homepage.

File: pegasus/sites.v3/hourofcode.com/public/index.haml (hunk 7 above)
Comment (tanyaparker) [reply pegasus/sites.v3/hourofcode.com/public/index.haml:228]: Stakeholders want this design as the evergreen design, even after Hour of Code. They've agreed that we have so many CTA at the top and they're okay with having the video permanently "downgraded" to next to the sign up form.

**Comment by tanyaparker** [comment]
@breville @Erin007 I've adjusted the styles a bit after some feedback with Mark. Sharing some updated screenshots. PTAL

Brendan - The map page didn't really change. Both the homepage and /map use the `hoc_events_counter` view. They previously used it the same way, as a title above the map. But now the styles have diverged and I'm showing that I've applied the styles on the separate pages instead of in the view so that the string content can be used on both pages but look different.

| Desktop | Tablet | Mobile |
| --------- | ------- | ------ |
| screencapture-localhost-hourofcode-3000-us-2018-10-23-23_55_29 | screencapture-localhost-hourofcode-3000-us-2018-10-23-23_55_52 | screencapture-localhost-hourofcode-3000-us-2018-10-23-23_56_18 |

**320px width**
I think the wonkiness of the language dropdown is just because of Chrome not displaying that right in their mobile preview. When I shrink the browser width manually the dropdown displays fine.
image

File: pegasus/sites.v3/hourofcode.com/public/css/front-page.css [hunk 8]
```diff
@@ -179,10 +215,18 @@ a.category-link {
   text-decoration: none;
 }
 
-.watch-video-small {
-  color: white;
-  cursor: pointer;
-  font-size: 16px;
+.video-responsive {
+  overflow: hidden;
+  padding-bottom: 56.25%;
```
Comment (breville) [inline pegasus/sites.v3/hourofcode.com/public/css/front-page.css]: This is the most specific number I've ever seen :)  Comment on what it means?

**Pull request merged**